src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -192,6 +196,27 @@ private function validateGatewayNonce(string $gatewayNonce)
         }
     }
 
+    /**
+     * Validate Donation Form Status
+     *
+     * @unreleased
+     */
+    private function validateDonationFormStatus(int $formId)
+    {
+        $donationForm = DonationForm::find($formId);
+
+        if (!$donationForm || $donationForm->status->isTrash()) {
+            wp_die(
+                esc_html__(
+                    'This donation form is not accepting donations.',
+                    'give'
+                ),
+                esc_html__('Error', 'give'),
+                ['response' => 403]
```

Comment:
Can the admin still process a donation through a Trashed form? I think they can still see them, so does it make sense to donation to them?

If not, then I think the `403` makes sense, otherwise we should return a `401` if not an admin.

Reply:
@kjohnson nope, trashed forms are not accessible in the admin or front-end - so doesn't matter if authenticated or not. The only way would be donating programmatically, or if you are donating to a published form & it becomes trashed in the time you click submit.